BoxContentSDK/BoxContentSDK/Requests/BOXFileDeleteRequest.m

```diff
@@ -42,52 +63,109 @@ - (BOXAPIOperation *)createOperation
+            [JSONoperation.APIRequest setValue:self.matchingEtag forHTTPHeaderField:BOXAPIHTTPHeaderIfMatch];
+        }
+        
+        return JSONoperation;
     }
-
-    return JSONoperation;
 }
 
 - (void)performRequestWithCompletion:(BOXErrorBlock)completionBlock
 {
-    BOOL isMainThread = [NSThread isMainThread];
-    BOXAPIJSONOperation *folderOperation = (BOXAPIJSONOperation *)self.operation;
-    
-    folderOperation.success = ^(NSURLRequest *request, NSHTTPURLResponse *response, NSDictionary *JSONDictionary) {
-
-        if ([self.cacheClient respondsToSelector:@selector(cacheFileDeleteRequest:error:)]) {
-            [self.cacheClient cacheFileDeleteRequest:self error:nil];
-        }
-        if (completionBlock) {
-            [BOXDispatchHelper callCompletionBlock:^{
-                completionBlock(nil);
-            } onMainThread:isMainThread];
-        }
-    };
-    folderOperation.failure = ^(NSURLRequest *request, NSHTTPURLResponse *response, NSError *error, NSDictionary *JSONDictionary) {
-
-        if ([self.cacheClient respondsToSelector:@selector(cacheFileDeleteRequest:error:)]) {
-            [self.cacheClient cacheFileDeleteRequest:self error:error];
-        }
-        if (completionBlock) {
-            [BOXDispatchHelper callCompletionBlock:^{
-                completionBlock(error);
-            } onMainThread:isMainThread];
-        }
-    };
-    [self performRequest];
+    if ([self shouldPerformBackgroundOperation] == YES) {
+        BOOL isMainThread = [NSThread isMainThread];
+        
+        BOXAPIDataOperation *fileOperation = (BOXAPIDataOperation *)self.operation;
+        __weak BOXAPIDataOperation *weakFileOperation = fileOperation;
+        
+        fileOperation.successBlock = ^(NSString *modelID, long long expectedTotalBytes) {
+            NSData *data = [NSData dataWithContentsOfFile:weakFileOperation.destinationPath];
+            
+            if ([self.cacheClient respondsToSelector:@selector(cacheFileDeleteRequest:error:)]) {
+                [self.cacheClient cacheFileDeleteRequest:self error:nil];
+            }
+            
+            if ([[NSFileManager defaultManager] fileExistsAtPath: weakFileOperation.destinationPath]) {
+                NSError *error = nil;
+                [[NSFileManager defaultManager] removeItemAtPath:weakFileOperation.destinationPath error:&error];
+            }
+            
+            if (completionBlock) {
+                [BOXDispatchHelper callCompletionBlock:^{
+                    completionBlock(nil);
+                } onMainThread:isMainThread];
+            }
+        };
+        
+        fileOperation.failureBlock = ^(NSURLRequest *request, NSHTTPURLResponse *response, NSError *error) {
+            if ([self.cacheClient respondsToSelector:@selector(cacheFileDeleteRequest:error:)]) {
+                [self.cacheClient cacheFileDeleteRequest:self error:error];
+            }
+            if (completionBlock) {
```

Comment:
Do we not have to delete the destination file here? Is it not created if the server returns an error code, even if the server response has a body?

Reply:
Ok, we have it right. No destination file is created when we have an api error. Only created for success.